packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -692,6 +701,38 @@ class TextSelectionOverlay {
 
   void _handleAnyDragEnd(DragEndDetails details) => _selectionOverlay.hideMagnifier(shouldShowToolbar: !_selection.isCollapsed);
 
+  // The center of the handle is where it's easier to drag, and
+  // [details.globalPosition] is probably close to center.
+  // Return offset from center of handle to center of correct text line.
+  // The offset use to shift real drag point on handle to text position point.
+  // Make sure even if the drag events happened on other position
+  // of handle, not center. The text position point always locate in
+  // correct text line.
+  Offset _getOffsetToTextPositionPoint(TextSelectionHandleType type) {
+    final Size handleSize = selectionControls!.getHandleSize(
+      renderObject.preferredLineHeight,
+    );
+
+    // Try to shift center of handle to top by half of handle height.
+    final double halfHandleHeight = handleSize.height / 2;
+
+    // [getHandleAnchor] use to shift selection end point to left top point
+    // of handle rect when build handle widget.
+    // End point is at bottom of selection rect, which is at the bottom of
+    // the text line too.
+    // Try to shift handle top to selection end point by dy of handle anchor.
```

Comment:
[getHandleAnchor] is used to shift the selection endpoint to the top left point of the handle rect when building the handle widget.  The endpoint is at the bottom of the selection rect, which is also at the bottom of the line of text.  Try to shift the top of the handle to the selection endpoint by the dy of the handle's anchor.

Reply:
Done.